tests/core/process.c

```diff
@@ -117,9 +125,31 @@ static gboolean malloc_range_check_cb (
     const GumMallocRangeDetails * details, gpointer user_data);
 #endif
 
+GThread * new_sleeping_dummy_thread_sync (volatile gboolean * done)
+{
+  struct _TestThreadSyncData sync_data;
+  GThread * thread;
+
+  g_cond_init (&sync_data.thread_is_started);
+  g_mutex_init (&sync_data.thread_is_started_mutex);
+  g_mutex_lock (&sync_data.thread_is_started_mutex);
+  sync_data.done = done;
+
+  thread = g_thread_new ("process-test-sleeping-dummy", sleeping_dummy,
+      &sync_data);
+
+  g_cond_wait (&sync_data.thread_is_started,
```

Comment:
Spurious wake-ups are possible, so one should always wait on a variable to change. (Look at how `g_cond_wait()` is used elsewhere.)